metrics/web/docs/analysis/deep_links/01_deep_links_analysis.md

```diff
@@ -0,0 +1,244 @@
+
+    return DashboardPageDeepLinks(selectedProjectGroup: selectedProjectGroup);
+  }
+}
+
+/// A [ChangeNotifier] that provides an ability to manage the application's deep links.
+class DeepLinksNotifier extends ChangeNotifier {
+
+  // Other fields, constructor here...
+
+  /// A current value of the deep link in the application.
+  PageDeepLinks _currentDeepLinks;
+
+  /// Handles the given [routeConfiguration] and updates the [PageDeepLinks] value.
+  void handleRouteConfiguration(RouteConfiguration routeConfiguration) {
+    // If the deep links are associated with a Dashboard page
+    if(routeConfiguration.name = RouteName.dashboard) {
+      final routeParameters = routeConfiguration.routeParameters;
+
+      _currentDeepLinks = DashboardPageDeepLinks.fromRouteParameters(routeParameters);
+      notifyListeners();
+    }
+    // Other routes handling...
+  }
+}
+```
+
+</details>
+
+#### Providing page-specific deep links to corresponding page presenters
+<details>
+  <summary>Code snippet</summary>
+
+```dart
+/// An abstract class that represents a [ChangeNotifier] of a specific page.
+abstract class PageNotifier extends ChangeNotifier {
+  /// Provides the current state of deep links associated with this page.
+  PageDeepLinks get pageDeepLinks;
+
+  /// Handles the deep links updates represented by the given
+  /// [pageDeepLinks].
+  void handleDeepLinkUpdates(PageDeepLinks pageDeepLinks);
+}
+
+/// A listener that triggers the [SomePageNotifier.handleDeepLinkUpdates] on [DeepLinksNotifier.currentDeepLinks]
+/// updates.
+void deepLinksNotifierListener() {
+  final currentDeepLinks = deepLinksNotifier.currentDeepLinks;
+
+  somePageNotifier.handleDeepLinkUpdates(currentDeepLinks);
+}
+```
+
+</details>
+
+#### Handling page-specific deep links updates provided by page presenters
+<details>
+  <summary>Code snippet</summary>
+
+```dart
+/// A listener that triggers the [DeepLinksNotifier.handlePageDeepLinks] on [SomePageNotifier.pageDeepLinks]
+/// updates.
+void somePageNotifierListener() {
+  final pageDeepLinks = somePageNotifier.pageDeepLinks;
+
+  deepLinksNotifier.handlePageDeepLinks(pageDeepLinks);
+}
+```
+
+</details>
+
+#### Providing deep links' updates to the `NavigationNotifier`
+<details>
+  <summary>Code snippet</summary>
+
+```dart
+/// A listener that triggers the [NavigationNotifier.saveDeepLinks] on [DeepLinksNotifier.currentDeepLinks]
+/// updates.
+void deepLinksNotifierListener() {
+  final currentDeepLinks = deepLinksNotifier.currentDeepLinks;
+
+  navigationNotifier.saveDeepLinks(currentDeepLinks);
+}
+```
+</details>
+
+### System modeling
+> Create an abstract model of the system/feature.
+
+Consider the following class diagram that describes how the feature can be implemented:
+![Class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/dashboard/raw/deep_links_analysis/metrics/web/docs/analysis/deep_links/diagrams/deep_links_class_diagram.puml)
```

Comment:
It seems like the class diagram is out of scope for the analysis document. What do you think about removing it from here?